src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,41 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    public static final int FIGURE_COUNT = 5;
+
+    public static Figure getRandomFigure() {
+        Random random = new Random();
+        int randomChoice = random.nextInt(FIGURE_COUNT); // Choose a random type of figure
+        String color = ColorSupplier.getRandomColor();
+
+        switch (randomChoice) {
+            case 0:
+                double squareSide = random.nextDouble() * 10 + 1;
+                return new Square(squareSide, color);
+            case 1:
+                double length = random.nextDouble() * 10 + 1;
+                double width = random.nextDouble() * 10 + 1;
+                return new Rectangle(length, width, color);
+            case 2:
+                double base = random.nextDouble() * 10 + 1;
+                double height = random.nextDouble() * 10 + 1;
+                return new RightTriangle(base, height, color);
+            case 3:
+                double radius = random.nextDouble() * 10 + 1;
+                return new Circle(radius, color);
+            default:
+                double topBase = random.nextDouble() * 10 + 1;
+                double bottomBase = random.nextDouble() * 10 + 1;
+                double trapezoidSide = random.nextDouble() * 10 + 1;
+                return new IsoscelesTrapezoid(topBase, bottomBase, trapezoidSide, color);
```

Comment:
1) This part has a problem with code duplication. You generate random sizes using one formula. Let's imagine tomorrow you'll need to change this formula, then you'll have to go over each line and make a change. It's insufficient. In this case, you can create a separate method for this computation and reuse it.
2) Magic numbers 10 and 1. You should make them constants as these numbers are not supposed to be changed during the program run. https://mate-academy.github.io/jv-program-common-mistakes/java-core/abstract-class-interface/oop-advanced#all-magic-numbers-in-your-code-should-be-constants